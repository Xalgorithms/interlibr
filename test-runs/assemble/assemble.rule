META
  VERSION "0.0.1"
  MAINTAINER "Don Kelly <karfai@gmail.com>";

REQUIRE assemble:origin:0.0.1 AS table_origin;
REQUIRE assemble:join:0.0.1 AS table_join;

ASSEMBLE table0
  COLUMNS FROM table_origin;

ASSEMBLE table1
  COLUMNS (a, c) FROM table_origin
  COLUMN p FROM table_join;
