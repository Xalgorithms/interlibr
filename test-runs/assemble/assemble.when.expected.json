{
  "tables" : {
    "table0" : [
      { "a" : 1, "b": 2, "c" : 3 },
      { "a" : 11, "b": 22, "c" : 33 }
    ],
    "table1" : [
      { "a" : 1, "c" : 3, "p" : "p0" },
      { "a" : 11, "c" : 33, "p" : "p0" }
    ],
    "table2" : [
      { "a" : 1,  "c" : 3, "d" : 66 },
      { "a" : 11, "c" : 33          }
    ]
  }
}
